src/V3Param.cpp

```diff
@@ -190,8 +190,8 @@ class ParameterizedHierBlocks final {
         // hierOptParamp width.
         V3Number varNum(pinValuep, hierOptParamp->num().width());
         if (hierOptParamp->isDouble()) {
-            varNum.isDouble(true);
             if (pinValuep->isDouble()) {
+                varNum.dataType(V3NumberDataType::DOUBLE);
```

Comment:
I think V3NumberDataType should become private, and we should avoid any external usages changing the internal type.  I suggest make a separate pull we can pre-merge that here instead constructs a V3Number as double, or constructs as a non-double.

Reply:
Fixed in https://github.com/verilator/verilator/pull/3529